Automattic/wp-calypso
# Sites Management Page: Improve render performance by paginating when greater than 96 sites

Fixes #67612

## Proposed Changes

* Applies pagination when displaying greater than 96 sites in order to improve render performance. Uses `96` instead of `100` as the default `perPage` to better support 3 and 2 column rows.
* Exports `handleQueryParamChange` function for use by `<SitesDashboard />` component.

<img width="1347" alt="image" src="https://user-images.githubusercontent.com/36432/189738393-ff4dd278-b164-4cf9-9b37-e42d06829f8d.png">

## Testing Instructions

1. Navigate to `/sites` with a WordPress user that has greater than 3 sites.
2. Add `?per-page=3` to the URL.
3. Observe the pagination appear at the bottom of the sites list.
4. Click on the pagination component buttons and observe expected behavior (paginating through the results).

One note: when manipulating `?per-page=3`, you may end up in this stacked component scenario:

<img width="1378" alt="image" src="https://user-images.githubusercontent.com/36432/189739015-86924661-f7af-4aa3-a8d0-ab024d4bfb30.png">

While stacking the pagination and hidden sites components doesn't look like the optimal design, I don't think it's really worth solving for because it's a really uncommon scenario.
